rust/rope/src/tree.rs

```diff
@@ -790,6 +808,27 @@ impl<'a, N: NodeInfo> Cursor<'a, N> {
     }
 }
 
+/// An iterator generated by a [`Cursor`], for some [`Metric`].
+pub struct CursorIter<'c, 'a: 'c, N: 'a + NodeInfo, M: 'a + Metric<N>> {
+    cursor: &'c mut Cursor<'a, N>,
+    __metric: PhantomData<&'a M>,
```

Comment:
Why two underlines? I think one is standard. I could be wrong though.

Reply:
No better reason than, "I've seen this elsewhere". Looking into this I guess I've only seen that on things that have to be `pub` for some reason? will fix.